### PR TITLE
Feature: Skybox Resource & Importer

### PR DESCRIPTION
**Pull request by Bernatmago**
This is an intermediate step for the new skybox. For now it uses 6 resource skybox (1 per face) to separat it from the textures pipeline.

Adressing @miquelmiro3 . I know it seems stupid to separate it from texture resources but it will make more sense once we have hdr skybox. Even if the importing is very similar we need to make a distinction between the resources types and keep them in their respective locations and im pretty certain the hdr version will have a diferent internal resource structure. The idea is to convert it into a single resource skybox with all what it needs inside

It also fixes the skybox hack with y orientation and faces i did because i was getting a weird skybox using texture resources

### CODE REVIEW

**Review by monikamnc — APPROVED** [review]
Everything is coc

File: Source/src/importers/SkyboxImporter.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    public:
+        SkyboxImporter() = default;
+        ~SkyboxImporter() override = default;
+
+        void Import(const char* path, YAML::Node& meta) override;
+        Resource* Load(UID id) override;
+
+        void Save(UID id, const Resource* resource) override;
+
+    private:
```
Comment (armandojaga) [inline Source/src/importers/SkyboxImporter.h:20]:
```suggestion
```

File: Source/src/modules/ModuleResources.cpp [hunk 2]
```diff
@@ -177,6 +183,7 @@ std::vector<UID> Hachiko::ModuleResources::CreateAsset(Resource::Type type, cons
 void Hachiko::ModuleResources::LoadAsset(const std::string& path)
 {
     Resource::AssetType asset_type = GetAssetTypeFromPath(path);
+
```
Comment (armandojaga) [inline Source/src/modules/ModuleResources.cpp:186]:
```suggestion
```

File: Source/src/resources/ResourceSkybox.cpp [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+#include "ResourceSkybox.h"
+
+Hachiko::ResourceSkybox::ResourceSkybox() : Resource(Resource::Type::SKYBOX) {}
+
+Hachiko::ResourceSkybox::ResourceSkybox(UID id) : Resource(id, Resource::Type::SKYBOX) {}
+
+Hachiko::ResourceSkybox::~ResourceSkybox()
+{
+    if (id != 0)
```
Comment (armandojaga) [inline Source/src/resources/ResourceSkybox.cpp:11]:
```suggestion
    if (id != 0) {
```

File: Source/src/resources/ResourceSkybox.cpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        glDeleteTextures(1, &id);
```
Comment (armandojaga) [inline Source/src/resources/ResourceSkybox.cpp:12]:
```suggestion
        glDeleteTextures(1, &id);
    }
```

**Review by armandojaga — CHANGES_REQUESTED** [review]

**Review by armandojaga — APPROVED** [review]

**Pull request merged**